index.bs

```diff
@@ -12337,18 +12336,77 @@ The value of the [=function object=]’s <code class="idl">name</code> property
+    such as "|object| is an |interface| object", to mean, "|object| implements |interface|".
+</div>
+
 Every [=platform object=] is associated with a global environment, just
 as the [=initial objects=] are.
 It is the responsibility of specifications using Web IDL to state
 which global environment (or, by proxy, which global object) each platform
 object is associated with.
 
+<div algorithm>
+    To <dfn>create an object implementing the interface</dfn> |interface|, with optional
+    intenal slots |slots|, for an interface which is not declared with the
+    [{{Global}}] [=extended attribute=], perform the following steps:
+
+    1. If |slots| is provided, append \[[PrimaryInterface]] to |slots|.
+    1. Otherwise, let |slots| be &laquo; \[[PrimaryInterface]] &raquo;.
+    1. Let |proto| be the [=interface prototype object=] of |interface|.
+    1. Let |instance| be [$ObjectCreate$](|proto|, |slots|).
+    1. Set |instance|.\[[PrimaryInterface]] to |interface|.
+    1. Let |realm| be the current [=Realm=].
+    1. [=Define the unforgeable regular operations=] of |interface| on |instance|, given |realm|.
+    1. [=Define the unforgeable regular attributes=] of |interface| on |instance|, given |realm|.
+    1. Return |instance|.
+</div>
+
+<div class="note">
+    Specifications use various wordings to reference the "[=create an object implementing
+    the interface=]" algorithm, such as "a new |interface| object".
+</div>
+
+<div class="note">
+
+    The set of interfaces that a [=platform object=] [=implements=] does not change over
+    the lifetime of the object.
+
+    Multiple [=interface objects=] associated with different [=Realm/global objects=]
+    will allocate [=platform objects=] which have \[[PrimaryInterface]] internal
+    slots referencing the same underlying [=interface=]. For example, a page may
+    contain a same-origin iframe, with the iframe's method being called on the
+    main page's element of the same kind, with no exception thrown.
+
+    [=Interface mixins=] do not participate directly in the evaluation of the [=implements=]
+    algorithm. Instead, each [=interface=] that the [=interface mixin=] is [=included=] in
+    has its own "copy" of each [=member=] of the [=interface mixin=], and the corresponding
+    [=creating an operation function|operation function=] checks that the receiver [=implements=]
+    the particular [=interface=] which [=includes=] the [=interface mixin=].
+
+</div>
+
 The <dfn id="dfn-primary-interface" export>primary interface</dfn> of a platform object
-that implements one or more interfaces is the most-derived [=interface=]
-that it implements.  The value of the \[[Prototype]] [=internal slot=]
+that [=implements=] one or more interfaces is the value of the object's
+\[[PrimaryInterface]] internal slot, which is is the most-derived [=interface=]
+that it [=implements=].  The value of the \[[Prototype]] [=internal slot=]
```

Comment:
Except when subclassing is involved.... I'm not sure there's much of a point to this sentence about [[Prototype]] now.  It's neither normative nor very informative.

Reply:
Good point, deleted.